devito/interfaces.py

```diff
@@ -270,13 +269,11 @@ def __init__(self, *args, **kwargs):
             self.indices = self._indices(**kwargs)
             self.dtype = kwargs.get('dtype', np.float32)
             self.space_order = kwargs.get('space_order', 1)
-            initializer = kwargs.get('initializer', None)
-            if initializer is not None:
-                assert(callable(initializer))
-            self.initializer = initializer
+            self.initializer = kwargs.get('initializer', None)
+            if self.initializer is not None:
+                assert(callable(self.initializer))
             self.numa = kwargs.get('numa', False)
-            self._data = kwargs.get('_data', None)
```

Comment:
Well, not really, `initializer` is effectively some kernel, `_data` is a raw pointer. Anyway, not a showstopper, just something to keep in mind...